junit-common/src/main/java/org/jboss/weld/junit/common/AbstractWeldInitiator.java

```diff
@@ -0,0 +1,379 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.weld.junit.common;
+
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.NormalScope;
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.event.Event;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.InjectionTarget;
+import javax.enterprise.util.TypeLiteral;
+
+import org.jboss.weld.config.ConfigurationKey;
+import org.jboss.weld.environment.ContainerInstance;
+import org.jboss.weld.environment.se.Weld;
+import org.jboss.weld.environment.se.WeldContainer;
+import org.jboss.weld.inject.WeldInstance;
+
+/**
+ *
+ * @author <a href="mailto:manovotn@redhat.com">Matej Novotny</a>
+ */
+public abstract class AbstractWeldInitiator implements WeldInstance<Object>, ContainerInstance {
+
+    /**
+     * The returned {@link Weld} instance has:
+     * <ul>
+     * <li>automatic discovery disabled</li>
+     * <li>concurrent deployment disabled</li>
+     * </ul>
+     *
+     * @return a new {@link Weld} instance suitable for testing
+     */
+    public static Weld createWeld() {
+        return new Weld().disableDiscovery().property(ConfigurationKey.CONCURRENT_DEPLOYMENT.get(),
+            false);
+    }
+
+    protected final Weld weld;
+
+    // not final so that we can automatically add there the test intance for JUnit 5
+    protected List<ToInject> instancesToInject;
```

Comment:
Intentionally not `final`, we need a way to add the current test instance as injection target and this has to be done separately from `WeldInitiator.Builder` as user might have built that already.

Reply:
`final` does not mean immutable. You can safely make this field final.